protoc-gen-gogo/generator/generator.go

```diff
@@ -3366,6 +3384,64 @@ func (g *Generator) generateExtensionRegistration(ext *ExtensionDescriptor) {
 	}
 }
 
+func (g *Generator) generateResetField(message *Descriptor, field *descriptor.FieldDescriptorProto) {
+	fieldName := g.GetFieldName(message, field)
+
+	// You could do "for key := range m.fieldName { delete(m.fieldName) }" potentially
+	// But not even sure if this keeps the same map with the same capacity
+	// And it could be very slow
+	if g.IsMap(field) {
+		g.P(`m.`, fieldName, ` = nil`)
+		return
+	}
+
+	if field.IsRepeated() {
+		g.P(`if m.`, fieldName, ` != nil {`)
+		g.In()
+		g.P(`m.`, fieldName, ` = m.`, fieldName, `[:0]`)
+		g.Out()
+		g.P(`}`)
+		return
+	}
+
+	if field.IsRequired() || gogoproto.NeedsNilCheck(message.proto3(), field) {
+		g.P(`m.`, fieldName, ` = nil`)
+		return
+	}
+
+	var zeroValue string
+	switch *field.Type {
+	case descriptor.FieldDescriptorProto_TYPE_DOUBLE,
+		descriptor.FieldDescriptorProto_TYPE_FLOAT:
+		zeroValue = "0.0"
```

Comment:
Is this distinction from other numbers necessary? I figured this would be an untyped literal.

Reply:
Yea I don't know, it's probably fine to just do `"0"`, but hey if we're doing the switch then why not.